feat: Implement Chatting component

choiseona opened the pull request:
## 📂 작업 내용

closes #27 

- [x] 작업 내용

## 💡 자세한 설명

- props로 채팅 내용 받기
- ChatBubble 공통 컴포넌트 사용
- Input 공통 컴포넌트 사용

## 📗 참고 자료 & 구현 결과 (선택)

<img width="248" alt="image" src="https://github.com/user-attachments/assets/ab030297-445c-4c47-b599-2001a132a4dc">

## 📢 리뷰 요구 사항 (선택)
공통 컴포넌트가 아니어서 components/main 폴더 생성 후 만들었는데, 좀 더 좋은 폴더명이 있을까요??🤔

## 🚩 후속 작업 (선택)

## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 불필요한 코드는 제거했나요?

choiseona committed:
feat: Add Chatting component for message interaction

- Create Chatting component with message display and input
- Support different message types (user/others) with ChatBubble
- Add responsive layout with proper spacing

choiseona committed:
design: Swap ChatBubble variant between user and others messages

- Switch variant='secondary' from others to user messages

choiseona commented on client/src/components/main/Chatting.tsx:
```diff
@@ -0,0 +1,35 @@
+        여기에다가 답하고
+        <br /> 채팅할 수 있습니다.
+      </p>
+      <div className="flex w-full flex-col gap-2">
+        {messages.map((message: Message) => {
+          const { nickname, content } = message;
+          if (message.isOthers) return <ChatBubble content={content} nickname={nickname} />;
```
헉 key 추가하겠습니다.

choiseona replied:
추가했습니닷

choiseona committed:
fix: Add missing key prop

- Add unique key prop to mapped elements for React rendering optimization

choiseona committed:
design: Make Chatting component width flexible

- Remove fixed max-width constraint
- Allow component to adapt to container width

choiseona committed:
design: Add focus style to Input component

- Add violet border color on input focus
- Remove default focus outline

choiseona committed:
design: Limit chat bubble width for better readability

- Set chat bubble max-width to 85% of container
- Improve visual distinction between messages

choiseona committed:
design: Limit chat bubble width for better readability

- Set chat bubble max-width to 85% of container
- Improve visual distinction between messages

rhino-ty commented on client/src/components/main/Chatting.tsx:
```diff
@@ -0,0 +1,36 @@
+  messages: Message[];
+}
+
+const Chatting = ({ messages, className, ...props }: ChattingProps) => {
```
선아님! 이 컴포넌트 `main` 폴더가 아니고 `ui` 폴더로 가는 건 아닐까요? 아니면 도메인별로 나누기로 했으니 `chat`폴더에 둬도 되구요

choiseona replied:
chat 폴더로 이동시키겠습니다~!

choiseona replied:
<img width="145" alt="image" src="https://github.com/user-attachments/assets/ef29d400-94cf-4393-8672-b36a930f780e">

choiseona committed:
chore: Create chat domain folder and reorganize related files

- Created a chat folder to organize chat-related functionality.
- Moved relevant files into the new folder
